syncer/ddl.go

```diff
@@ -33,25 +40,22 @@ type parseDDLResult struct {
 	isDDL  bool
 }
 
-func (s *Syncer) parseDDLSQL(sql string, p *parser.Parser, schema string) (result parseDDLResult, err error) {
-	sql = utils.TrimCtrlChars(sql)
+type shardingDDLInfo struct {
+	name       string
+	tableNames [][]*filter.Table
+	stmt       ast.StmtNode
+}
 
-	// check skip before parse (used to skip some un-supported DDLs)
-	ignore, err := s.skipQuery(nil, nil, sql)
-	if err != nil {
-		return parseDDLResult{
-			stmt:   nil,
-			ignore: false,
-			isDDL:  false,
-		}, err
-	} else if ignore {
+// filterDDLSQL filter two type of sql: 1. tidb not support 2. not ddl sql such as begin/flush.
+func (s *Syncer) filterDDLSQL(sql string, p *parser.Parser) (result parseDDLResult, err error) {
+	sql = utils.TrimCtrlChars(sql)
+	if ignore := utils.IsBuildInSkipDDL(sql); ignore || sql == "BEGIN" {
```

Comment:
`sql == "BEGIN"` here is duplicated with the first line of `handleQueryEvent`, we could
- keep that line, remove this check. Or
- remove the check in `handleQueryEvent`, but a benchmark is needed.